genie-agent/src/main/java/com/netflix/genie/agent/execution/services/impl/grpc/GRpcAgentJobServiceImpl.java

```diff
@@ -79,6 +82,43 @@
         this.jobServiceProtoConverter = jobServiceProtoConverter;
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void handshake(
+        final AgentClientMetadata agentClientMetadata
+    ) throws HandshakeException {
+        final HandshakeRequest request;
+        try {
+            request = jobServiceProtoConverter.toHandshakeRequest(agentClientMetadata);
+        } catch (final GenieConversionException e) {
+            throw new HandshakeException("Failed to construct request from parameters", e);
+        }
+
+        final HandshakeResponse response = handleResponseFuture(this.client.handshake(request));
+
+        switch (response.getType()) {
+            case ALLOWED:
+                log.info("Successfully shook hand with server");
+                break;
+
+            case REJECTED:
```

Comment:
Did you want to log something here or just relying on something upstream logging it?